connectors/byoei.py

```diff
@@ -127,6 +135,17 @@ async def _batch_bulk(self, operations):
                         if "error" in data:
                             logger.error(f"operation {op} failed, {data['error']}")
                             raise Exception(data["error"]["reason"])
+
+            for op in operations:
+                if OP_INDEX in op or OP_UPSERT in op:
+                    self.indexed_document_count += 1
+                elif OP_DELETE in op:
+                    self.deleted_document_count += 1
+                elif "doc" in op:  # update payload
+                    self.indexed_document_volume += get_byte_size(op["doc"])
+                else:  # index payload
+                    self.indexed_document_volume += get_byte_size(op)
```

Comment:
>I think this info is already present in the async bulk for every call we make

May I know where? I don't think we have the size of the documents indexed into ES.

Reply:
we don't have the details but we have one call here 

https://github.com/elastic/connectors-python/blob/main/connectors/byoei.py#L116

that get the size of the request.

If you are collecting docs size, doc per doc it's duplicated work. We would need to change the other call so it uses `self.indexed_document_volume` instead. it's ok if it's not taking into account the request wrap data.

Reply:
>it's ok if it's not taking into account the request wrap data

I think that should be OK? It doesn't have to be that accurate.

Reply:
I forgot about one more place, we do calculate the size for each doc. Another spot where you get the doc size (from `MemQueue`) is here https://github.com/elastic/connectors-python/blob/main/connectors/byoei.py#L142

if you can reuse that size it means you don't need to calculate it again.

Reply:
what you could do is use `bulk_size` at https://github.com/elastic/connectors-python/blob/main/connectors/byoei.py#L149


```
self.indexed_document_volume += bulk_size
```

Reply:
Hi @tarekziade , I didn't make it reusable, because:
1. the function `get_mb_size` is used in DEBUG logging ([1](https://github.com/elastic/connectors-python/blob/fa8a6ecb13f957b0c85e1fde53c1de840d4af592/connectors/byoei.py#L118-L120), [2](https://github.com/elastic/connectors-python/blob/fa8a6ecb13f957b0c85e1fde53c1de840d4af592/connectors/byoei.py#L470))
2. We could make the first one reusable, but I feel it's unnecessary, as this only adds one `asizeof` call (assuming `DEBUG` is turned on)

Reply:
This adds one call per document, it will raise the CPU usage and you already have that info when you dequeue

Reply:
>you already have that info when you dequeue

As I mentioned here https://github.com/elastic/connectors-python/pull/528#discussion_r1130736180, before this PR, the `get_mb_size` is only used in two places (DEBUG logging), and the first usage is to calculate the size of the whole `operations`:
https://github.com/elastic/connectors-python/blob/fa8a6ecb13f957b0c85e1fde53c1de840d4af592/connectors/byoei.py#L118-L120

From what we discussed before, we need to count the number/volume of docs based on the bulk api response, so we have to calculate the size of each document.